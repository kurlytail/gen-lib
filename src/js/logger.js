import winston from 'winston';
import { format } from 'logform';

const logger = winston.createLogger({
    transports: [new winston.transports.Console()],
    format: format.simple()
});

export default logger;
