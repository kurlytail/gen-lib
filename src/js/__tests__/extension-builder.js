import FS from 'fs';
import Extension from '../extension';
import ExtensionBuilder from '../extension-builder';
import logger from 'logger';

jest.mock('fs');
jest.mock('../extension');
jest.mock('logger');

const FIXTURES = {
    GENERATOR: {
        design: 'b',
        map: 'c',
        options: {
            extension: Array.apply(null, { length: 10 })
                .map(Number.call, Number)
                .map(n => n.toString() + '._')
        }
    },
    STATS_FILE: {
        isFile: () => true,
        isDirectory: () => false
    },
    DIR_FILE: {
        isFile: () => false,
        isDirectory: () => true
    },
    READDIR: Array.apply(null, { length: 3 })
        .map(Number.call, Number)
        .map(n => n.toString() + '._')
};

describe('# ExtensionBuilder', () => {
    beforeEach(() => {
        FS.readdirSync.mockReset();
        FS.statSync.mockReset();
        FS.statSync.mockReturnValue(FIXTURES.STATS_FILE);
    });

    describe('## constructor', () => {
        it('### should construct default builder', () => {
            const builder = new Extension(FIXTURES.GENERATOR);
            expect(builder).toMatchSnapshot();
        });
    });

    describe('## load', () => {
        it('### should load template from files and folders', () => {
            FS.readdirSync
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValueOnce(FIXTURES.READDIR)
                .mockReturnValue([]);
            FS.statSync
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValueOnce(FIXTURES.DIR_FILE)
                .mockReturnValue(FIXTURES.STATS_FILE);
            const builder = new ExtensionBuilder(FIXTURES.GENERATOR);
            builder.build();
            expect(builder).toMatchSnapshot();
        });
    });
});
