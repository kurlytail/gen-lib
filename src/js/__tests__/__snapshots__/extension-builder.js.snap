// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# ExtensionBuilder ## constructor ### should construct default builder 1`] = `
Extension {
  "load": [MockFunction],
  "toString": [MockFunction],
}
`;

exports[`# ExtensionBuilder ## load ### should load template from files and folders 1`] = `
ExtensionBuilder {
  "_extensions": Array [
    Object {
      "file": "0._/0._/0._",
      "name": "0",
    },
    Object {
      "file": "0._/0._/1._",
      "name": "1",
    },
    Object {
      "file": "0._/0._/2._",
      "name": "2",
    },
    Object {
      "file": "0._/1._",
      "name": "1",
    },
    Object {
      "file": "0._/2._",
      "name": "2",
    },
    Object {
      "file": "1._",
      "name": "1",
    },
    Object {
      "file": "2._",
      "name": "2",
    },
    Object {
      "file": "3._",
      "name": "3",
    },
    Object {
      "file": "4._",
      "name": "4",
    },
    Object {
      "file": "5._",
      "name": "5",
    },
    Object {
      "file": "6._",
      "name": "6",
    },
    Object {
      "file": "7._",
      "name": "7",
    },
    Object {
      "file": "8._",
      "name": "8",
    },
    Object {
      "file": "9._",
      "name": "9",
    },
  ],
  "_generator": Object {
    "design": "b",
    "map": "c",
    "options": Object {
      "extension": Array [
        "0._",
        "1._",
        "2._",
        "3._",
        "4._",
        "5._",
        "6._",
        "7._",
        "8._",
        "9._",
      ],
    },
  },
}
`;
